lib/generators/administrate/dashboard/dashboard_generator.rb

```diff
@@ -122,6 +126,20 @@ def klass
         @klass ||= Object.const_get(class_name)
       end
 
+      def class_path_filename_fragment
+        if @class_path.present?
+          "/#{@class_path.join('/')}"
+        end
+      end
```

Comment:
Will look into making a test for it

Reply:
This might also benefit from my advice about factoring out a difficult algorithm into a separate class, and then unit-testing the separate class. That way the generator spec doesn't need to get too complicated, and you still get to test edge cases.